src/test/java/seedu/address/model/session/SessionDateTest.java

```diff
@@ -14,4 +16,53 @@ public void constructor_null_throwsNullPointerException() {
         assertThrows(NullPointerException.class, () -> new SessionDate(dateValue, null));
         assertThrows(NullPointerException.class, () -> new SessionDate(null, timeValue));
     }
+
+    @Test
+    public void isValidSessionDate() {
+        String correctTimeValue = "23:59";
+
+        // checks for empty input
+        assertThrows(SessionException.class, () -> new SessionDate("", correctTimeValue));
+
+        // checks for invalid month
+        assertThrows(SessionException.class, () -> new SessionDate("2021-13-01", correctTimeValue));
+        assertThrows(SessionException.class, () -> new SessionDate("2021-00-01", correctTimeValue));
+
+        // checks for invalid day
+        assertThrows(SessionException.class, () -> new SessionDate("2021-12-00", correctTimeValue));
+        assertThrows(SessionException.class, () -> new SessionDate("2021-12-32", correctTimeValue));
+
+        // checks for leap year
+        assertThrows(SessionException.class, () -> new SessionDate("2021-02-29", correctTimeValue));
+
+        // checks for err in date format
+        assertThrows(SessionException.class, () -> new SessionDate("2021-13-001", correctTimeValue));
+        assertThrows(SessionException.class, () -> new SessionDate("2021-013-01", correctTimeValue));
+        assertThrows(SessionException.class, () -> new SessionDate("02021-13-01", correctTimeValue));
+        assertThrows(SessionException.class, () -> new SessionDate("20210-013-01", correctTimeValue));
+        assertThrows(SessionException.class, () -> new SessionDate("2021/13/01", correctTimeValue));
+        assertThrows(SessionException.class, () -> new SessionDate("2021.13.01", correctTimeValue));
+
+    }
+
+
+
+    @Test
+    public void isValidSessionTime() {
+        String correctDateValue = "2021-01-01";
+
+        // checks for empty input
+        assertThrows(SessionException.class, () -> new SessionDate(correctDateValue, ""));
+
+        // checks for correct time format
+        assertThrows(SessionException.class, () -> new SessionDate(correctDateValue, "24:00"));
+        assertThrows(SessionException.class, () -> new SessionDate(correctDateValue, "23:60"));
+        assertThrows(SessionException.class, () -> new SessionDate(correctDateValue, "30:99"));
+        assertThrows(SessionException.class, () -> new SessionDate(correctDateValue, "0000"));
+        assertThrows(SessionException.class, () -> new SessionDate(correctDateValue, "00-00"));
+        assertThrows(SessionException.class, () -> new SessionDate(correctDateValue, "00.00"));
+        assertThrows(SessionException.class, () -> new SessionDate(correctDateValue, "12pm"));
+        assertThrows(SessionException.class, () -> new SessionDate(correctDateValue, "12:52pm"));
+        assertThrows(SessionException.class, () -> new SessionDate(correctDateValue, "09:32am"));
+    }
```

Comment:
Actually just thinking what would happen if we put the time as 23:592. This is because when trying out the iP, I noticed that one of those that I am reviewing failed this test, so you could include 1 more test with this value to ensure that adding extra values to the timing will not cause any unexpected behavior ah.

Reply:
Ok, I will try adding 23:002 and 23:592. So to double confirm, 23:002 and 23:592 should fail, right?

Reply:
Yeap! I think that the test case should fail with these values.